tests/js/transformationsTests.js

```diff
@@ -0,0 +1,138 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+// Declare dependencies
+/* global JSON */
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.tests.chartAuthoring.transforms");
+
+    gpii.tests.chartAuthoring.transforms.assertTransformation = function (msg, model, transform, expected) {
+        var actual = fluid.model.transform(model, transform);
+        jqUnit.assertDeepEq(msg, expected, actual);
+    };
+
+    gpii.tests.chartAuthoring.transforms.stringToNumberTestCases = {
+        transform: {
+                transform: {
+                type: "gpii.chartAuthoring.transforms.stringToNumber",
+                inputPath: "value",
+                outputPath: "value"
+            }
+        },
+        models: [{value: "1"}, {value: "1.1"}, {value: 1}, {value: 1.1}, {value: ""}, {value: null}, {}],
+        expected: [{value: 1}, {value: 1.1}, {value: 1}, {value: 1.1}, {value: null}, {value: null}, {}]
+    };
+
+    jqUnit.test("Test gpii.chartAuthoring.transforms.stringToNumber", function () {
+        fluid.each(gpii.tests.chartAuthoring.transforms.stringToNumberTestCases.models, function (model, idx) {
+            var expected = gpii.tests.chartAuthoring.transforms.stringToNumberTestCases.expected[idx];
+            gpii.tests.chartAuthoring.transforms.assertTransformation("The model '" + JSON.stringify(model) + "' should have been transformed", model, gpii.tests.chartAuthoring.transforms.stringToNumberTestCases.transform, expected);
+        });
+    });
+
+    gpii.tests.chartAuthoring.transforms.percentageInputs = [undefined, null, NaN, false, true, function () {}, {}, ["array"], "", "string", 2.2, "2.2", 0, "0", 50, 100];
+    gpii.tests.chartAuthoring.transforms.percentageOutputs = [
```

Comment:
I think that these giant blocks of nulls are getting a little silly :) This also mostly duplicates gpii.tests.calculatePercentage.outputs in the other testing file.

I suggest that you hugely condense this table by i) just verifying that the numeric values are as expected (in the last 5x5 block of the table) , and then ii) writing a general driver that determines that the output is null in all cases where total or values are of the wrong types